resources/config-task.json

```
@@ -0,0 +1,22 @@
+{
+  "server": {
+    "storagePath": "data",
+    "cacheMaxMb": 1000
+  },
+  "connection": {
+    "storageMethod": "file",
+    "parallelism": 4,
+    "defaults": {
+      "index": {
+        "reindexMinBytes": 100000,
+        "reindexMaxBytes": 10000000,
+        "maxOldIndexes": 3
+      }
+    }
+  },
+  "task": {
+    "id": "migrate/sid",
+    "ledgers": ["ledger1", "ledger2"],
+    "force": false
+  }
```

I think this would be clearer if it was just:
```suggestion
  "sidMigration": {
    "ledgers": ["ledger1", "ledger2"],
    "force": false
  }
```

---

I actually started with that design, but if we add more tasks in the future then we'll have a proliferation of config schemas, whereas by nesting it under the "task" key we only need to adapt that one schema. If you feel strongly about it I can change it back, but I thought this would lead to a more stable abstraction.

---

I don't think there's a difference, but I could be wrong. The way it is now, we have to check specific arguments for each task with a different schema as we add more tasks, and we dispatch on the "id" key to decide which schema to use. I don't think we can expect to know the precise arguments for each new task type before they are defined.

Instead, I'm proposing that we configure specific modules in each system in their own sub-map. For example, what if people want to run multiple tasks at once? What about multiple tasks on different sets of ledgers?

Unless I'm missing something, each new task will add a new schema. The only difference is whether that new schema is validated directly or if all of the "task" schemas are grouped into a multi schema dispatched on "id", but there will be at least one schema for each task either way.

---

These I think are the important parts of a task abstraction:
- no need to add new integrant handlers
- localized schema updates
- defined order of execution

I like nesting the task config underneath the `task` keyword because it lets us reuse the integrant handler for every task.

I think both approaches are rather localized for schema updates, though if you want to define multiple tasks we'd either need a vector of task-maps in the style that I wrote or a vector of task-ids in the style you wrote. Or we could just elide that entirely and say you can only run one task at time and that will force sequential execution.

I do think we should try to get the abstraction right so users don't have to learn multiple incompatible approaches to running.

---

I think we should have a separate key for the migration, and any other commands like this we'll run because the configuration is clearer. "reusing the integrant handler" doesn't make any sense to me. Each migration, or "task" as you put it, will have an entry point. Either there is a case statement that we'll have to modify each time we need to add something, or we define a new method for a new key. There is no difference.

These "tasks" are independent components in the system, so they should get independent configurations, with independent initializers.

---

Should I just rip out the whole task abstraction then? I think it's useful but if you don't we can do it your way.

---

I took it out, and I see what you mean - the difference is dispatching on `ig/init-key` vs `fluree.server.task/run-task`, so they're basically the same. No task abstraction means slightly more config but a shallower call stack.

---

I think the task abstraction actually had more config. With tasks, you had a task config which included an id along side the config for that id. The config is flatter how it is now, and you just have the config for each task that lives in its own key; no extra task key, no extra id key, and no need for an extra dispatch in the config parser.